cardinal/message/manager.go

```diff
@@ -50,7 +54,7 @@ func (m *Manager) RegisterMessages(msgs ...types.Message) error {
 		}
 
 		// Register message
-		m.registeredMessages[msg.Name()] = msg
+		m.registeredMessagesByName[msg.Name()] = msg
```

Comment:
registeredMessageByType is not updated here, but it doesnt matter because we should remove registeredMessageByType (i left a comment directly on it on why we dont need it). Simple API -> one less thing to screw up